backend/src/index.ts

```diff
@@ -72,39 +77,75 @@ const processSourceBlocks = (target: Target) => async (source: Source) => {
     const targetApi = await createApi(config.targetChainUrl);
 
     const target = new Target({ api: targetApi, logger, state });
+    const master = getAccount(config.accountSeed);
 
-    const sources = await Promise.all(
-      config.sourceChains.map(async ({ url, parachains }) => {
-        const api = await createApi(url);
-        const chain = await api.rpc.system.chain();
-        const sourceSigner = getAccount(`${config.accountSeed}/${chain}`);
-        const paraSigners = parachains.map(({ paraId }) => getAccount(`${config.accountSeed}/${paraId}`));
-
-        // TODO: can be optimized by sending batch of txs
-        // TODO: master has to delegate spending to sourceSigner and paraSigners
-        const master = getAccount(config.accountSeed);
-        for (const delegate of [sourceSigner, ...paraSigners]) {
-          // send 1.5 units
-          await target.sendBalanceTx(master, delegate, 1.5);
+    if (args.length && (args[0] === 'archive')) {
+      if (!config.archives) {
+        throw new Error("Archives are not provided");
+      }
```

Comment:
This is never reachable as long as the type system is used properly.

Did you mean `!config.archives.length` instead?